src/modules/blockchainApplication/explore/components/BlockchainApplications/BlockchainApplications.test.js

```diff
@@ -1,44 +1,55 @@
 import { screen } from '@testing-library/react';
-import mockBlockchainApplications from '@tests/fixtures/blockchainApplicationsExplore';
 import { usePinBlockchainApplication } from '@blockchainApplication/manage/hooks/usePinBlockchainApplication';
-import { renderWithRouter } from 'src/utils/testHelpers';
+import { renderWithRouterAndQueryClient } from 'src/utils/testHelpers';
+import { useBlockchainApplicationExplore } from '../../hooks/queries/useBlockchainApplicationExplore';
+import { useBlockchainApplicationStatistics } from '../../hooks/queries/useBlockchainApplicationStatistics';
+import { mockBlockchainApp, mockBlockchainAppStatistics } from '../../__fixtures__';
 import BlockchainApplications from './BlockchainApplications';
 
-jest.useFakeTimers();
-jest.mock('@blockchainApplication/manage/hooks/usePinBlockchainApplication');
 const mockTogglePin = jest.fn();
-const mockedPins = [mockBlockchainApplications[0].chainID];
+const mockedPins = [mockBlockchainApp.data[0].chainID];
+const mockApplyFilters = jest.fn();
+const mockFetchNextPage = jest.fn();
+
+jest.useFakeTimers();
+jest.mock('../../../manage/hooks/usePinBlockchainApplication');
+jest.mock('../../hooks/queries/useBlockchainApplicationExplore');
+jest.mock('../../hooks/queries/useBlockchainApplicationStatistics');
+jest.mock('@common/hooks', () => ({
+  ...jest.requireActual('@common/hooks'),
+  useFilter: jest.fn(() => ({
+    applyFilters: mockApplyFilters,
+  })),
+}));
 
 usePinBlockchainApplication.mockReturnValue({
   togglePin: mockTogglePin,
   pins: mockedPins,
   checkPinByChainId: jest.fn().mockReturnValue(true),
 });
 
-describe('BlockchainApplicationList', () => {
-  const props = {
-    applyFilters: jest.fn(),
-    filters: jest.fn(),
-    applications: {
-      data: mockBlockchainApplications,
-      isLoading: true,
-      loadData: jest.fn(),
-      error: false,
-    },
-    statistics: {
-      data: {
-        registered: 101,
-        active: 53,
-        terminated: 9,
-        totalSupplyLSK: '5000000',
-        stakedLSK: '3000000',
-      },
-    },
-  };
+useBlockchainApplicationExplore.mockReturnValue({
```

Comment:
what you're gaining by mocking an API call? lets MSW do the job for you

Reply:
@soroushm MSW isn't working fully in a lot of cases including where we want different responses so we've been mocking custom hooks lately. We've raised the issue with you in the past and once it's resolved, we'll update such mocks